odbc/src/api/functions.rs

```diff
@@ -882,134 +964,132 @@ pub extern "C" fn SQLGetStmtAttrW(
-                    StatementAttribute::ParamsetSize => unsafe {
-                        *(value_ptr as *mut ULen) = stmt_contents.attributes.paramset_size;
-                    },
-                    StatementAttribute::QueryTimeout => unsafe {
-                        *(value_ptr as *mut ULen) = stmt_contents.attributes.query_timeout;
-                    },
-                    StatementAttribute::RetrieveData => unsafe {
-                        *(value_ptr as *mut RetrieveData) = stmt_contents.attributes.retrieve_data;
-                    },
-                    StatementAttribute::RowBindOffsetPtr => unsafe {
-                        *(value_ptr as *mut _) = stmt_contents.attributes.row_bind_offset_ptr;
-                        set_str_length(string_length_ptr, size_of::<*mut ULen>() as Integer)
-                    },
-                    StatementAttribute::RowBindType => unsafe {
-                        *(value_ptr as *mut ULen) = stmt_contents.attributes.row_bind_type;
-                    },
-                    StatementAttribute::RowNumber => unsafe {
-                        *(value_ptr as *mut ULen) = stmt_contents.attributes.row_number;
-                    },
-                    StatementAttribute::RowOperationPtr => unsafe {
-                        *(value_ptr as *mut _) = stmt_contents.attributes.row_operation_ptr;
-                        set_str_length(string_length_ptr, size_of::<*mut USmallInt>() as Integer)
-                    },
-                    StatementAttribute::RowStatusPtr => unsafe {
-                        *(value_ptr as *mut _) = stmt_contents.attributes.row_status_ptr;
-                        set_str_length(string_length_ptr, size_of::<*mut USmallInt>() as Integer)
-                    },
-                    StatementAttribute::RowsFetchedPtr => unsafe {
-                        *(value_ptr as *mut _) = stmt_contents.attributes.rows_fetched_ptr;
-                        set_str_length(string_length_ptr, size_of::<*mut ULen>() as Integer)
-                    },
-                    StatementAttribute::RowArraySize => unsafe {
-                        *(value_ptr as *mut ULen) = stmt_contents.attributes.row_array_size;
-                    },
-                    StatementAttribute::SimulateCursor => unsafe {
-                        *(value_ptr as *mut ULen) = stmt_contents.attributes.simulate_cursor;
-                    },
-                    StatementAttribute::UseBookmarks => unsafe {
-                        *(value_ptr as *mut UseBookmarks) = stmt_contents.attributes.use_bookmarks;
-                    },
-                    StatementAttribute::AsyncStmtEvent => unsafe {
-                        *(value_ptr as *mut _) = stmt_contents.attributes.async_stmt_event;
-                    },
-                }
-            }
-            SqlReturn::SUCCESS
+    let stmt = must_be_valid!(stmt_handle.as_statement());
+    if value_ptr.is_null() {
+        return SqlReturn::SUCCESS;
```

Comment:
Can we add a comment explaining why this is the return value in this case?

Reply:
I think it should actually be `ERROR`. That was left over from fixing segfaults on win64.